### ISSUE TITLE
Batching should respect repo merge strategy

### ISSUE THREAD

**Issue by brettdh** [smarkets/marge-bot#287]
I have a repo that's configured to use semi-linear history - that is, require fast-forward merges, but always add a merge commit. This works as expected with marge-bot when it merges one MR at a time, but when marge-bot batches MRs together, a couple odd things happen:

1. No per-MR merge commits appear on the target branch
2. Some MRs in the batch are marked "Closed" rather than "Merged"

I suspect (2) is related to (1) - the merged commits don't match the commits on the MRs, so Gitlab can't match them up and see that the MR was closed. Oddly, though, it was the opposite of what I'd expect:

- MR 1 has commits A, B, and C
- MR 2 has commit D
- The batch MR queues up A-B-C-D' (there may have been a rebase atop new changes on the target branch; I don't recall)
- CI passes; marge-bot merges the batch
- MR 1 is marked "closed"; MR 2 is marked "merged".

Sorry for the vagueness; I'll try to put together an actual demonstration soon.

**Comment by brettdh** [comment]
Hmm, I just came across #264, which is released in 0.9.3 but not documented in the README. I'll try this out to see if it resolves (1) above.

**Comment by yangm97** [comment]
Did you come to a conclusion? Now that I look into it, it does seem like #264 would address my issue #266 too, but I'm not sure.

**Comment by brettdh** [comment]
Nope, sorry; haven't looked at this in a while. I may have time a couple weeks from now; not sure.
